pkg/containers/containers.go

```diff
@@ -182,62 +202,98 @@ func (c *Containers) populate() error {
 // saving container information in Containers CgroupInfo map.
 // NOTE: ALL given cgroup dir paths are stored in CgroupInfo map.
 func (c *Containers) CgroupUpdate(cgroupId uint64, path string, ctime time.Time) (CgroupInfo, error) {
-	containerId, runtime := getContainerIdFromCgroup(path)
+	containerId, containerRuntime := getContainerIdFromCgroup(path)
+	container := cruntime.ContainerMetadata{
+		ContainerId: containerId,
+	}
 
 	info := CgroupInfo{
-		Path:        path,
-		ContainerId: containerId,
-		Runtime:     runtime,
-		Ctime:       ctime,
+		Path:      path,
+		Container: container,
+		Runtime:   containerRuntime,
+		Ctime:     ctime,
 	}
 
 	c.mtx.Lock()
 	c.cgroups[uint32(cgroupId)] = info
 	c.mtx.Unlock()
 
+	err := c.enrichCgroupInfo(cgroupId)
+
+	//don't fail on enrichment but log to stderr
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to enrich container data: %v", err)
+	}
+
 	return info, nil
 }
 
+// enrichCgroupInfo checks for a given cgroupId if it is relevant to some running container
+// it then calls the runtime info service to gather additional data from the container's runtime
+func (c *Containers) enrichCgroupInfo(cgroupId uint64) error {
+	c.mtx.RLock()
+	info := c.cgroups[uint32(cgroupId)]
+	c.mtx.RUnlock()
+	containerId := info.Container.ContainerId
+	runtime := info.Runtime
+
+	if containerId != "" && runtime != cruntime.Unknown {
+		metadata, err := c.enricher.Get(containerId, runtime, context.Background())
```

Comment:
You're always using context.Background(), is it really needed to have it as the 3rd argument then ?

Besides, shouldn't we create a timed-out context in here so we don't let ContainerEnricher implementing the interface to be stuck in Get() forever and ever ? (I'm not sure all APIs talking to container daemons will have timed-out attempts, etc).

Reply:
Then we should just use the provided Context argument here, seems like an oversight from me. Would you prefer if we hardcoded the context timeout logic?

Reply:
Sorry, I don't follow what you said. I thought either we remove the ctx argument OR created a timed-out context so enricher would not hang forever.

Reply:
I misread where this code comment was.
We have the the context as an argument exactly so such a change (like adding a timeout) could be done once here, instead of being done over and over in each implementation.
Since I had no decision on what to do in here before, I just passed the background context. Now I can simply add a timeout logic here.

Reply:
Done, please resolve if it's fine by you now.